src/clr-angular/forms/select/select.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {AfterContentInit, Component, ElementRef, ViewChild} from "@angular/core";
+
+import {POPOVER_HOST_ANCHOR} from "../../popover/common/popover-host-anchor.token";
+import {IfOpenService} from "../../utils/conditional/if-open.service";
+
+@Component({
+    selector: "clr-select",
+    templateUrl: "select.html",
+    providers: [IfOpenService, {provide: POPOVER_HOST_ANCHOR, useExisting: ElementRef}],
+    host: {"[class.clr-select]": "true"}
+})
+export class ClrSelect implements AfterContentInit {
+    @ViewChild("input") input: ElementRef;
+
+    constructor(private ifOpenService: IfOpenService) {}
+
+    private registerPopoverIgnoredInput() {
+        if (this.input) {
+            this.ifOpenService.registerIgnoredElement(this.input);
+        }
+    }
+
+    onClick(event: MouseEvent): void {
+        this.ifOpenService.toggleWithEvent(event);
+        if (this.input) {
+            this.input.nativeElement.focus();
```

Comment:
I think you have a bug if the menu is open and you click on the caret again. The click will force re-focus on the input over and over.

Reply:
My thinking here was that even if you click on the caret the `input` should be focused because the native select too works the same way, no? In the native select, the trigger doesn't not function separately. Let me know what you think about this @youdz.

Reply:
You're right, the focus should stay focused after closing the menu.
But as we saw during our call, we still have a focus/open/close bug.